### DISCUSSION TITLE
Problem creating workload item when logged in as non-admin

### DISCUSSION THREAD

**Discussion by box-turtle** [microsoft/Microsoft-Fabric-workload-development-sample#30] [Help] [answered]
We just started having an issue in our dev environment where non-admin users are not able to create our 3rd party workload items. The error we receive from the create item call POST to

```
https://wabi-us-north-central-j-primary-redirect.analysis.windows.net/metadata/workspaces/58326055-de45-4e9c-80ac-689c94c96a8f/artifacts


Activityid: 95c4d20d-2e34-4825-8c36-ea5ec25bca71
Requestid: 538afd12-07e1-1fdc-8ae1-a43a69487759


```

is not very informative - we get a 404 with this response body:

```
{
    "error": {
        "code": "PriorityPlacementCoreServiceNotFound",
        "pbi.error": {
            "code": "PriorityPlacementCoreServiceNotFound",
            "parameters": {},
            "details": [],
            "exceptionCulprit": 1
        }
    }
}
```

I CAN create an item when logged in as a Fabric admin user, it's only normal users who have this difficulty. The strange thing is that they did not have a problem with this last week.

This only affects our 3rd party workload - users are able to create regular synapse notebooks and other items in that capacity.

If I am logged in as a normal user and navigate to the workspace where I normally create items, and then check the workspace settings/ license info, it tells me the workspace is assigned to a Fabric capacity, but it says `This workspace is configured to use a capacity you don't have access to `

<img width="862" alt="image" src="https://github.com/user-attachments/assets/faa679f7-0646-4e47-8bbf-18266307844c">

**Comment by dhrumil83** [comment]
I am getting a similar error creating new items.
Url: https://wabi-west-us-redirect.analysis.windows.net/metadata/workspaces/4e313df9-54aa-4510-a864-bcc9825793b5/artifacts

```
{
    "error": {
        "code": "PriorityPlacementCoreServiceNotFound",
        "pbi.error": {
            "code": "PriorityPlacementCoreServiceNotFound",
            "parameters": {},
            "details": [],
            "exceptionCulprit": 1
        }
    }
}
```

**Reply by anitamayorov** [reply]
Hi @box-turtle and @dhrumil83,

Can you please elaborate on the scenario - are you trying to create an item for developer scenario using a developer workload connected via DevGateway or is this a 'test' scenario with an activated workload for tenant? 
Did this non-admin user have access to that capacity last week?

**Reply by dhrumil83** [reply]
For me it's deployed at the tenant: "test" scenario with activated workload.

It works fine via DevGateway. At one point get yesterday, create and get item was working, and now all of a sudden this error.
I have not tried this via a Fabric admin account yet.

Note: My user is capacity admin.

**Reply by box-turtle** [reply]
Hi @anitamayorov 

It is my understanding the only way to engage development mode is to use an admin user - correct? We do not see the problem logging in as an admin and running dev mode / devgateway.

Our scenario is in a test scenario with an activated workload for the tenant.

I did not check the workspace settings and see that message last week because everything was working as expected - the user could create/save items as expected. The problem only started yesterday.

And when I say non-admin user I mean all the non-admin users we had added to this tenant, it's not just a single user.

One other detail is that these users are "guest users" who were not created on that tenant but added as guests. If you look at the permissions it shows that they have "contributor" access to the capacity. It is only if you navigate to the workspace settings > license info screen that you see that message `This workspace is configured to use a capacity you don't have access to`.

**Reply by ramizsha** [reply]
@box-turtle Do you maybe have a dev instance registered (devgateway running)? if so, it might be interfering with the test scenario if the workloads have the same name - try disconnecting it and try to create an item again, if it fails, please provide another requestId so I can take a look at it from my end.

**Reply by ramizsha** [reply]
@dhrumil83 can you please provide a requestId so I can take a look at the issue from my end? you can find it in the response headers of the https call.

**Reply by box-turtle** [reply]
> @box-turtle Do you maybe have a dev instance registered (devgateway running)? if so, it might be interfering with the test scenario if the workloads have the same name - try disconnecting it and try to create an item again, if it fails, please provide another requestId so I can take a look at it from my end.

No devgateway is running. Closed browser and started a new session this morning -same result, a 404 error on the request 

```
https://wabi-us-north-central-j-primary-redirect.analysis.windows.net/metadata/workspaces/add277b2-6fd2-4664-b967-7a8968b055a4/artifacts
```

with response

```
{
    "error": {
        "code": "PriorityPlacementCoreServiceNotFound",
        "pbi.error": {
            "code": "PriorityPlacementCoreServiceNotFound",
            "parameters": {},
            "details": [],
            "exceptionCulprit": 1
        }
    }
}
```

Request Id

```
Requestid: 283bd208-660c-8823-07e5-4bfdd988e776
```

**Reply by ramizsha** [reply]
We have identified an issue with the garbage collection of dev instances. It doesn't cause issues when working in dev scenario but it may cause issues when working on the same capacity both in dev and in test/prod scenario. Until we fix this issue you can mitigate it by using separate capacities for dev and prod/test scenarios. Our recommendation is to use a capacity for test or prod that wasn't used untill now for dev scenario.

**Reply by box-turtle** [reply]
@ramizsha - 

We created a new capacity in the same tenant and attached a workspace to that capacity.

Now when trying to create a workload item in that workspace I receive a 500 error rather than a 404 error.

The create item call:

```
https://wabi-us-north-central-j-primary-redirect.analysis.windows.net/metadata/workspaces/add277b2-6fd2-4664-b967-7a8968b055a4/artifacts
```

The request id:

```
d8183da3-6af7-6bb6-2b66-cabb24863d17
```

The returned error:

```
{
    "error": {
        "code": "UnknownError",
        "pbi.error": {
            "code": "UnknownError",
            "parameters": {},
            "details": []
        }
    }
}
```

Oddly, if I switch the workspace back to the previous capacity (the one we had used with devgateway) and try to create an item, I get the same 404 error I was getting earlier. 

Then if I switch back to the new capacity I get the 500 error shown above.

**Reply by box-turtle** [reply]
Just to update further, now I can't create an item on the original capacity using dev mode/devgateway.exe - I get a 500 error trying to create or load items.

I'm basically dead in the water.

Request id for create call using dev mode/devgateway in the original capacity:

```
78f0818f-e3c5-99ed-f85c-6d530d6b84d3
```

**Reply by ramizsha** [reply]
@box-turtle two points:
1. Regarding this request Id "d8183da3-6af7-6bb6-2b66-cabb24863d17", I see that the request to your workload returns 404, the request we send is a post request that looks like this: <your BE endpoint>/workspaces/<workspaceobjectid>/items/<itemType>/<itemobjectid>. can you validate that this url is reachable?
2. Regarding this request Id "78f0818f-e3c5-99ed-f85c-6d530d6b84d3", I see that in this flow you are working with the same capacity used in the above request ("d8183da3-6af7-6bb6-2b66-cabb24863d17").
Please make sure to switch to a workspace that is assigned to the same capacity you registered your dev instance on.

**Reply by box-turtle** [reply]
> @box-turtle two points:
> 
> 1. Regarding this request Id "d8183da3-6af7-6bb6-2b66-cabb24863d17", I see that the request to your workload returns 404, the request we send is a post request that looks like this: /workspaces//items//. can you validate that this url is reachable?

Request with id `d8183da3-6af7-6bb6-2b66-cabb24863d17` was the one where I used a workspace with a **newly created capacity** and it returned a `500` error, not a `404`. 

> 2. Regarding this request Id "78f0818f-e3c5-99ed-f85c-6d530d6b84d3", I see that in this flow you are working with the same capacity used in the above request ("d8183da3-6af7-6bb6-2b66-cabb24863d17").
>    Please make sure to switch to a workspace that is assigned to the same capacity you registered your dev instance on.

I thought I had switched to a workspace assigned to the "old" capacity for this test (`d8183da3-6af7-6bb6-2b66-cabb24863d17`), but I must have made a mistake. I will retest the devgateway scenario.

**Reply by ramizsha** [reply, marked as answer]
> > @box-turtle two points:
> > 
> > 1. Regarding this request Id "d8183da3-6af7-6bb6-2b66-cabb24863d17", I see that the request to your workload returns 404, the request we send is a post request that looks like this: /workspaces//items//. can you validate that this url is reachable?
> 
> Request with id `d8183da3-6af7-6bb6-2b66-cabb24863d17` was the one where I used a workspace with a **newly created capacity** and it returned a `500` error, not a `404`.
> 
I understand, what I'm saying is that the request **to your workload** from Fabric returned 404 (In control plane APIs Fabric calls your workload).
> > 2. Regarding this request Id "78f0818f-e3c5-99ed-f85c-6d530d6b84d3", I see that in this flow you are working with the same capacity used in the above request ("d8183da3-6af7-6bb6-2b66-cabb24863d17").
> >    Please make sure to switch to a workspace that is assigned to the same capacity you registered your dev instance on.
> 
> I thought I had switched to a workspace assigned to the "old" capacity for this test (`d8183da3-6af7-6bb6-2b66-cabb24863d17`), but I must have made a mistake. I will retest the devgateway scenario.

**Reply by box-turtle** [reply]
ah - I see. Thank you. Let me do more checking on my end.

**Reply by box-turtle** [reply]
OK - we had a bad deployment of our BE code. Issues seem resolved now.

Thanks for all the help.

**Reply by dhrumil83** [reply]
> @dhrumil83 can you please provide a requestId so I can take a look at the issue from my end? you can find it in the response headers of the https call.

I changed a few admin settings and uploaded a new nuget file, and the error went away.
Not sure what was it, but everything seems to work now.
